docs/language-server/protocol-language-server.md

```diff
@@ -717,16 +724,17 @@ may be expanded in future.
  * @param creationTime creation time
  * @param lastAccessTime last access time
  * @param lastModifiedTime last modified time
- * @param kind type of [[FileSystemObject]], can be:
- * `Directory`, `File`, `Other`
+ * @param kind type of [[FileSystemObject]], can be: `Directory`, `File`, `Other`
  * @param byteSize size in bytes
+ * @param checksum: The SHA3-224 checksum of the file or directory contents.
  */
 interface FileAttributes {
   creationTime: UTCDateTime;
   lastAccessTime: UTCDateTime;
   lastModifiedTime: UTCDateTime;
   kind: FileSystemObject;
   byteSize: number;
+  checksum: SHA3-224;
```

Comment:
In the new API, writing and reading from a file requires knowing the file lenght. The only way to get the file length right now is to call `file/info` that returns `FileAttributes`. Computing checksum every time would be an overhead (especially for large files). I'd create a separate `file/checksum` request returning an SHA checksum.